### PR TITLE
feat: support smbus2 as drop-in replacement for I2C

### LINKED ISSUE #89: smbus2 support
Any chance a pull request that allows usage of `smbus2` as a drop-in replacement would be accepted? Potentially also as a fallback when `smbus` can not be loaded (or as a favored in case `smbus2` is installed). I am trying to make RPLCD work on Python 3 with balena and I am having a hard time getting the smbus packages in the right state.

### PR DESCRIPTION
**Pull request by joscha**
closes #89

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -54,6 +54,12 @@ If you want to use I²C, you also need smbus::
 
     $ sudo apt install python-smbus
 
+or alternatively `smbus2<https://pypi.org/project/smbus2/>`_::
+
+    $ sudo pip install smbus2
+
+RPLCD will first try to use smbus if available and if not, fall back to smbus2.
```
Comment (dbrgn) [inline README.rst]: To make the README a bit more compact, you could remove this section and change the one above it as follows:

```
If you want to use I²C, you also need either the smbus or `smbus2 <https://pypi.org/project/smbus2/>`_ library::

    $ sudo apt install python-smbus
    or
    $ sudo pip install smbus2

RPLCD will first try to use smbus if available and if not, fall back to smbus2.
```

**Comment by joscha** [comment]
changes in @dbrgn

**Comment by dbrgn** [comment]
Great, thank you :slightly_smiling_face:

**Comment by joscha** [comment]
\o/ any chance for a release with this change? Cheers!

**Comment by dbrgn** [comment]
Yeah, I'll try to remember to do one tonight :slightly_smiling_face: 

By the way, I noticed that in the tests I'm using `smbus-cffi`. That might have worked too, as it seems to be a CFFI based pure-python smbus binding with the same package name. But I guess with `smbus2` you're better off.

@joscha did you already test whether the `smbus2` based RPLCD works on an actual LCD?

**Comment by joscha** [comment]
> By the way, I noticed that in the tests I'm using `smbus-cffi`. That might have worked too

ah, I missed that, sorry.

> @joscha did you already test whether the smbus2 based RPLCD works on an actual LCD?

I tried in a fork via `git+https://github.com/joscha/RPLCD.git@joscha/smbus2#egg=RPLCD`, but would only get:
```
Traceback (most recent call last):
  File "src/main.py", line 1, in <module>
    from RPLCD.i2c import CharLCD
  File "/usr/local/lib/python3.6/site-packages/RPLCD/i2c.py", line 28, in <module>
    from .lcd import BaseCharLCD
  File "/usr/local/lib/python3.6/site-packages/RPLCD/lcd.py", line 27, in <module>
    from . import codecs
ImportError: cannot import name 'codecs'
```
I put that down to some problem with the git-based packaging.I've successfully used Matt Hawkin's script: https://bitbucket.org/MattHawkinsUK/rpispy-misc/raw/master/python/lcd_i2c.py with smbus2 instead of smbus and that worked, so I was assuming RPLCD would do the same. If you are worried you could do a named pre-release that I can test with?

**Comment by dbrgn** [comment]
Hm, yeah, I would have assumed that this works...

I'll simply put out the next version, and in the unlikely event that something breaks we can always create a patch release :slightly_smiling_face:

**Comment by dbrgn** [comment]
Version 1.2.0 is released. Testing would be very welcome!

If everything went right, you should now also have the `rplcd-tests` script available when installing the package via pip.

**Comment by joscha** [comment]
works a charm, tried via the testsuite 👍

**Comment by dbrgn** [comment]
Great to hear, thanks for testing @joscha :slightly_smiling_face:

**Pull request merged**